api-example.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Make sure to start the web UI with the following flags:
+#
+# python server.py --model MODEL --listen --no-stream
+
+# If no arguments were passed, display a usage message and exit with an error code
+if [[ "$#" -eq 0 ]]; then
+    echo "Usage: $0 [preset] prompt"
+    exit 1
+fi
+
+# Set LANG variable to ensure consistent language environment across different systems
+LANG=
+
+# Load preset
+if [[ -e "presets/$1.txt" ]]; then
+    source "presets/$1.txt"
+    shift
+fi
+
+# Set generation parameters
+prompt=$(jq -n --arg prompt "$*" '$prompt')
+max_new_tokens=${max_new_tokens:-200}
+do_sample=${do_sample:-true}
+temperature=$(printf "%.2f" "${temperature:-0.5}")
+top_p=$(printf "%.2f" "${top_p:-0.9}")
+typical_p=$(printf "%.2f" "${typical_p:-1}")
+repetition_penalty=$(printf "%.2f" "${repetition_penalty:-1.05}")
+encoder_repetition_penalty=$(printf "%.2f" "${encoder_repetition_penalty:-1.0}")
+top_k=${top_k:-0}
+min_length=${min_length:-0}
+no_repeat_ngram_size=${no_repeat_ngram_size:-0}
+num_beams=${num_beams:-1}
+penalty_alpha=$(printf "%.2f" "${penalty_alpha:-0}")
+length_penalty=$(printf "%.2f" "${length_penalty:-1}")
+early_stopping=${early_stopping:-false}
+seed=${seed:--1}
+
+# Encode data as JSON string
+data="{
+  \"data\": [
+    $(jq -n --arg prompt "$*" '$prompt'),
+    $max_new_tokens,
+    ${do_sample,,},
+    $temperature,
+    $top_p,
+    $typical_p,
+    $repetition_penalty,
+    $encoder_repetition_penalty,
+    $top_k,
+    $min_length,
+    $no_repeat_ngram_size,
+    $num_beams,
+    $penalty_alpha,
+    $length_penalty,
+    ${early_stopping,,},
+    $seed
+  ]
+}"
+
+# Make HTTP request and extract generated text
+response=$(curl -s -X POST "http://${server:-127.0.0.1}:7860/run/textgen" \
```

Comment:
```
${server:-127.0.0.1}
```
This is a little hard to read for me, but my bash isn't great.

I don't see `server` defined anywhere, should it be?  If this could be more explicit, i think it would be more readble.

Reply:
This is great - I've been looking for a way to run inference independently of the webui.  I think this is going to need to be extended pretty quickly, to allow for different character loading etc, but for now, super!  The port and server IP are important things to include (imo)

Also, I find the filename to be a little confusing... Can we call it `api-usage-example`?  there are other files callled `api-example` which are involved in standing up the api, but this is using that api.  Another idea for the name might be `inference.sh` or something like that.

When I tried working with the `/run/textgen` endpoint, i had some trouble, so having this available is fantastic.

Reply:
About the name: Since I created it by converting the existing `api-example.py` to bash, I kept the name, but you're right and it's more than just an example by now. I think it should start with `api-` to show it's relation to the other api-scripts. How about simply `api-bash-script`?

About the server variable: It uses an existing "server" variable and defaults to 127.0.0.1 when undefined. You can set it by prefixing it to the script call (just the way bash works), e. g.:

```
server=127.0.0.1 bash ./api-example.sh "$preset" "$prompt"
```

The word `bash` can be left out if you made the script executable (`chmod +x api-example.sh`).

Reply:
This has now been resolved by renaming the script and giving usage information and examples when it's run without arguments.